test/gtest/ucp/test_ucp_am.cc

```diff
@@ -1087,24 +1086,44 @@ class test_ucp_am_nbx_send_copy_header : public test_ucp_am_nbx {
     std::string m_hdr_copy;
 };
 
-UCS_TEST_P(test_ucp_am_nbx_send_copy_header, all_protos)
+UCS_TEST_SKIP_COND_P(test_ucp_am_nbx_send_copy_header, all_protos,
+                     (has_transport("self") && RUNNING_ON_VALGRIND))
 {
-    const unsigned random_iterations = 100;
+    /**
+     * Self transport is relying on memory copy for sending messages.
```

Comment:
using memcpy does not explain why need to skip - ib transport also uses memcpy for am_bcopy